source/PlayerInfo.cpp

```diff
@@ -1356,7 +1366,27 @@ void PlayerInfo::SetShipOrder(const vector<shared_ptr<Ship>> &newOrder)
 	// Check if the incoming vector contains the same elements
 	if(std::is_permutation(ships.begin(), ships.end(), newOrder.begin()))
 	{
+		auto OldFirstShip = ships[0];
+		int pos = 0;
 		ships = newOrder;
+		// Check if the Position of the Flagship has changed
+		if((ships[0] != OldFirstShip) && Preferences::Has("Automatically unpark flagship"))
+		{
+			// Find the position of the previous flagship in the new vector
+			for(long unsigned int i = 0; i < ships.size(); i++)
+			{
+				if(ships[i] == OldFirstShip)
+				{
+					pos = i;
+				}
+			}
```

Comment:
Maybe just use `std::find` and then subtract `begin()`.

Reply:
I didnt know that i can do it that way. This is awesome, thanks for the hint! :)